### DISCUSSION TITLE
Inconsistent port numbers (when native=True)

### DISCUSSION THREAD

**Discussion by xiaozhiagi** [zauberzeug/nicegui#1457] [Q&A] [answered]
### Question

The following is the ui.run configuration, which opens NiceGUI ready to go on at runtime http://localhost:8000 Instead of port 3005

```python

ui.run(
    title="dsp",
    # favicon="favicon.ico",
    storage_secret='THIS_NEEDS_TO_BE_CHANGED',
    host='127.0.0.1',
    port=3005,
    native=True,
    reload=False
)

```

**Comment by rodja** [comment, marked as answer]
Yes. The assumption is that you do not care about the port in native mode. Instead of using the port parameter, we actively search for an open port so you native app is starting even if the port is blocked:

https://github.com/zauberzeug/nicegui/blob/cc8c5d04b59ed71a01f3425c764fdc32aca428cb/nicegui/run.py#L138

**Reply by orennir** [reply]
@rodja I've just stumbled upon this issue also. We use native mode, but we also have custom FastAPI endpoints we use to communicate with other services. If we cannot know the port beforehand, how can we use those endpoints?

**Reply by rodja** [reply]
Interesting scenario. We would be happy to look at any pull request which uses the port parameter instead of finding an open port when running in native mode. Maybe you can use https://nicegui.io/documentation#urls to build a workaround? Depending on your use case you can publish the url on which the app is available somewhere.
